src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseCodeReviewFeedbackService.java

```diff
@@ -0,0 +1,239 @@
+
+        try {
+
+            setIndividualDueDateAndLockRepository(participation, programmingExercise, false);
+            this.programmingMessagingService.notifyUserAboutNewResult(automaticResult, participation);
+            // now the client should be able to see new result
+
+            log.debug("Submission id: {}", submission.getId());
+
+            var athenaResponse = this.athenaFeedbackSuggestionsService.getProgrammingFeedbackSuggestions(programmingExercise, (ProgrammingSubmission) submission, false);
+
+            List<Feedback> feedbacks = athenaResponse.stream().filter(individualFeedbackItem -> individualFeedbackItem.filePath() != null)
+                    .filter(individualFeedbackItem -> individualFeedbackItem.description() != null).map(individualFeedbackItem -> {
+                        var feedback = new Feedback();
+                        String feedbackText;
+                        if (Objects.nonNull(individualFeedbackItem.lineStart())) {
+                            if (Objects.nonNull(individualFeedbackItem.lineEnd()) && !individualFeedbackItem.lineStart().equals(individualFeedbackItem.lineEnd())) {
+                                feedbackText = String.format(NON_GRADED_FEEDBACK_SUGGESTION + "File %s at lines %d-%d", individualFeedbackItem.filePath(),
+                                        individualFeedbackItem.lineStart(), individualFeedbackItem.lineEnd());
+                            }
+                            else {
+                                feedbackText = String.format(NON_GRADED_FEEDBACK_SUGGESTION + "File %s at line %d", individualFeedbackItem.filePath(),
+                                        individualFeedbackItem.lineStart());
+                            }
+                            feedback.setReference(String.format("file:%s_line:%d", individualFeedbackItem.filePath(), individualFeedbackItem.lineStart()));
+                        }
+                        else {
+                            feedbackText = String.format(NON_GRADED_FEEDBACK_SUGGESTION + "File %s", individualFeedbackItem.filePath());
+                        }
+                        feedback.setText(feedbackText);
+                        feedback.setDetailText(individualFeedbackItem.description());
+                        feedback.setHasLongFeedbackText(false);
+                        feedback.setType(FeedbackType.AUTOMATIC);
+                        feedback.setCredits(0.0);
+                        return feedback;
+                    }).toList();
+
+            automaticResult.setSuccessful(true);
+            automaticResult.setCompletionDate(ZonedDateTime.now());
+
+            this.resultService.storeFeedbackInResult(automaticResult, feedbacks, true);
+
+            this.programmingMessagingService.notifyUserAboutNewResult(automaticResult, participation);
+        }
+        catch (Exception e) {
+            log.error("Could not generate feedback", e);
+            automaticResult.setSuccessful(false);
```

Comment:
To me it seems like you are abusing the `successful` flag here. This flag is meant to describe if the entire result is positive or not (i.e. all feedback entries are positive), not what the status of the feedback generation is. This could easily break code in other places and I strongly suggest that you think of another way to convey this. 
This could also confuse the user, as it would show successful in the middle of their (possibly not successful) other results that were created by the tests.

Reply:
I partially agree with the abuse of the flag.
The issue I wanted to address is whether this specific result is completed.
Regarding the user, in the UI, it's clearly stated that these are automatic AI results; they do not affect grading and are used only to enhance the learning process.
Ideally, I would add a new entity - feedback request - and on the client side, I would change the result-history component to support both the result and the feedback requests, separating them - but within the scope of my thesis and the next PR.